### ISSUE TITLE
[Bug] 2.16.4: java.lang.Long cannot be cast to java.lang.Integer

### ISSUE THREAD

**Issue by david-allison** [ankidroid/Anki-Android#14348]
```
java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
	at android.app.SharedPreferencesImpl.getInt(SharedPreferencesImpl.java:302)
	at org.acra.startup.LimiterStartupProcessor.processReports(SourceFile:55)
	at org.acra.startup.StartupProcessorExecutor.processReports$lambda$6$lambda$5(SourceFile:104)
	at org.acra.startup.StartupProcessorExecutor.a(Unknown Source:0)
	at n.a.run(Unknown Source:6)
	at java.lang.Thread.run(Thread.java:764)
```

https://ankidroid.org/acra/app/1/bug/70522/report/c5932e9a-3d95-4282-8dd9-85fbf968aff6

**Comment by david-allison** [comment]
@mikehardy this looks like https://github.com/ACRA/acra/issues/1233, fixed in https://github.com/ACRA/acra/commit/c193e35d6c2e4337dfc0b1d0090d28c915ba4ea1, tagged as [acra-5.11.1](https://github.com/ACRA/acra/releases/tag/acra-5.11.1)

This should have gone into AnkiDroid with d7082677bd8434fcffffc2e0339eeeda28652221, but something is fishy

**Comment by david-allison** [comment]
I have very limited time here

Current possibility: is the code in https://github.com/ACRA/acra/commit/c193e35d6c2e4337dfc0b1d0090d28c915ba4ea1 catching the Kotlin `ClassCastException`, and does this also handle `java.lang.ClassCastException`

**Comment by mikehardy** [comment]
That does look like the issue you linked, and I just did my best to check through the code (looked in the APK, looked in the source code tag that was built, checked web UI here), I came to the same conclusion you did. We appear to be using ACRA 5.11.1 already in the version of our app that crashed. Very strange!

**Comment by mikehardy** [comment]
Seems we have the opposite of the upstream original problem, *somehow* (I don't understand it, to be honest...) but the upstream issue is reopened and I've proposed a maybe-good-enough PR. Anyway, it's got eyes on it

**Comment by mikehardy** [comment]
FIxed upstream in 5.11.2 🥂

**Comment by mikehardy** [comment]
🤔 hmm - doesn't seem to be showing up from dependabot (yet?) and either way the gradle files have changed pretty radically now. Needs a specific bump on main and on release-2.16 branch I believe
